adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/AbstractPolicyEnforcer.java

```diff
@@ -309,7 +310,15 @@ private MethodConfig getRequiredScopes(PathConfig pathConfig, Request request) {
         MethodConfig methodConfig = new MethodConfig();
 
         methodConfig.setMethod(request.getMethod());
-        methodConfig.setScopes(pathConfig.getScopes());
+        List<String> scopes = new ArrayList<>(pathConfig.getScopes());
+
+        if (Boolean.TRUE.equals(getEnforcerConfig().getHttpMethodAsScope())) {
+            if (scopes.isEmpty()) {
```

Comment:
Since this still loads all of the scopes associated with the resource as defined in the server, you would never get an empty scopes list. You have to associate the scopes with the resource in order to be able to define a scope based permission (unless something else has changed.) So I think that the code should be:

List<String> scopes = new ArrayList<>();

if (Boolean.TRUE.equals(getEnforcerConfig().getHttpMethodAsScope())) {
    scopes.add(request.getMethod());
} else {
    scopes.addAll(pathConfig.getScopes());
}

Reply:
You are right. I've fixed and also updated test to make sure we behave as expected when resources are associated with scopes.